cirq-core/cirq/sim/sparse_simulator_test.py

```diff
@@ -1434,3 +1434,22 @@ def test_unseparated_states_str():
 qubits: (cirq.LineQubit(0), cirq.LineQubit(1))
 output vector: 0.707j|00⟩ + 0.707j|10⟩"""
     )
+
+
+@pytest.mark.parametrize('split', [True, False])
+def test_measurement_preserves_phase(split: bool):
+    c1, c2, t = cirq.LineQubit.range(3)
+    circuit = cirq.Circuit(
+        cirq.H(t),
+        cirq.measure(t, key="t"),
+        cirq.CZ(c1, c2).with_classical_controls("t"),
+        cirq.reset(t),
+    )
+    simulator = cirq.Simulator(split_untangled_states=split)
```

Comment:
I'd rather we tested `factor_state_vector` here directly. If you'd like to keep this test, then it should live closer to the simulator code.